File: algorithms/sorting/selectionSort/selection-sort.js
function selectionSort(array) {
    const size = array.length;
    for(let i = 0; i < size - 1; ++i) {
        let minIndex = i;
        for(let j = i + 1; j < size; ++j) {
            if(array[j] < array[minIndex]) {
                minIndex = j;
            }
        }
        if (array[i] !== array[minIndex]) {
        [array[i], array[minIndex]] = [array[minIndex], array[i]];
        }
    }
    return array;
}

File: algorithms/sorting/selectionSort/selection-sort.c
#include <stdio.h>

void selectionSort(int arr[], int size) {
    for (int i = 0; i < size - 1; i++) {
        int minIdx = i;
        for (int j = i + 1; j < size; j++) {
            if (arr[j] < arr[minIdx]) {
                minIdx = j;
            }
        }
        if (arr[i] != arr[minIdx]) {
            int tmp = arr[i];
            arr[i] = arr[minIdx];
            arr[minIdx] = tmp;
        }
    }
}

int main() {
    int arr[] = {5, 6, 9, 8, 7, 5};
    int size = sizeof(arr) / sizeof(arr[0]);

    selectionSort(arr, size);

    for (int i = 0; i < size; i++) {
        printf("%d, \n", arr[i]);
    }

    return 0;
}